code/modules/mining/mine_turfs.dm

```diff
@@ -32,27 +32,25 @@
 
 /turf/unsimulated/mineral/New()
 	. = ..()
-	MineralSpread()
 
-	spawn(1)
-		var/turf/T
-		if((istype(get_step(src, NORTH), /turf/simulated/floor)) || (istype(get_step(src, NORTH), /turf/space)) || (istype(get_step(src, NORTH), /turf/simulated/shuttle/floor)))
-			T = get_step(src, NORTH)
-			if (T)
-				T.overlays += image('icons/turf/walls.dmi', "rock_side_s")
-		if((istype(get_step(src, SOUTH), /turf/simulated/floor)) || (istype(get_step(src, SOUTH), /turf/space)) || (istype(get_step(src, SOUTH), /turf/simulated/shuttle/floor)))
-			T = get_step(src, SOUTH)
-			if (T)
-				T.overlays += image('icons/turf/walls.dmi', "rock_side_n", layer=6)
-		if((istype(get_step(src, EAST), /turf/simulated/floor)) || (istype(get_step(src, EAST), /turf/space)) || (istype(get_step(src, EAST), /turf/simulated/shuttle/floor)))
-			T = get_step(src, EAST)
-			if (T)
-				T.overlays += image('icons/turf/walls.dmi', "rock_side_w", layer=6)
-		if((istype(get_step(src, WEST), /turf/simulated/floor)) || (istype(get_step(src, WEST), /turf/space)) || (istype(get_step(src, WEST), /turf/simulated/shuttle/floor)))
-			T = get_step(src, WEST)
-			if (T)
-				T.overlays += image('icons/turf/walls.dmi', "rock_side_e", layer=6)
-	/*
+	if(ticker)
+		initialize()
+
+/turf/unsimulated/mineral/initialize()
+	MineralSpread()
+	for(var/dir in cardinal)
+		var/turf/T = get_step(src, dir)
+		if(T && (istype(T, /turf/simulated/floor) || istype(T, /turf/space) || istype(T, /turf/simulated/shuttle/floor)))
+			switch(dir)
+				if(NORTH)
+					T.overlays += image('icons/turf/walls.dmi', "rock_side_s")
```

Comment:
Images are pretty fucking cheap you know...

Icon operations is where the lag comes from!

<sup><sup>TG, beating dead performance horses since 2012</sup></sup>